src/EditorFeatures/Test2/NavigableSymbols/NavigableSymbolsTest.vb

```diff
@@ -0,0 +1,73 @@
+﻿' Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+Imports System
+Imports System.Linq
+Imports System.Threading
+Imports System.Threading.Tasks
+Imports Microsoft.CodeAnalysis.Editor.Host
+Imports Microsoft.CodeAnalysis.Editor.NavigableSymbols
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.BraceMatching
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Utilities
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Utilities.GoToHelpers
+Imports Microsoft.CodeAnalysis.Editor.UnitTests.Workspaces
+Imports Microsoft.CodeAnalysis.Navigation
+Imports Microsoft.VisualStudio.Composition
+Imports Microsoft.VisualStudio.Text
+Imports Roslyn.Test.Utilities
+Imports Xunit
+
+Namespace Microsoft.CodeAnalysis.Editor.UnitTests.NavigableSymbols
+
+    Public Class NavigableSymbolsTest
+
+        Private Shared ReadOnly s_exportProvider As ExportProvider = MinimalTestExportProvider.CreateExportProvider(
+            TestExportProvider.EntireAssemblyCatalogWithCSharpAndVisualBasic.WithParts(
+                GetType(MockDocumentNavigationServiceProvider),
+                GetType(MockSymbolNavigationServiceProvider)))
+
+        <WpfFact>
+        Public Async Function TestCharp() As Task
+            Using workspace = TestWorkspace.CreateCSharp("
+class C
+{
+    C$$ c
```

Comment:
Actually, there are two spans of interest:

1. the span that will be underlined in the editor.
2. teh location ew navigate to. 

you can use ```{|Whatever:...|}``` test spans to mark each.